any.hpp

```diff
@@ -221,20 +221,20 @@ struct static_any
 		using NonConstT = std::remove_cv_t<std::remove_reference_t<_T>>;
 		NonConstT* non_const_t = const_cast<NonConstT*>(&t);
 
-		std::array<char, _N> buff;
+		static_any temp = *this;
```

Comment:
isn't this expensive? will this call copy ctor of the stored object?
Wouldn't it be cheaper to not do it? provide weaker exception guarantee?

Maybe the copy could be replaced with move?

Reply:
expensive: it depends - yes it will call the copy ctor of the stored object. this is just fixing the bug by doing what we wanted to do, it keeps the strong guarantee yes.

I was already working on another version in a second branch, this is just the fix. I can do much better by just checking the type with is_trivially_copyable/is_nothrow_copy_assignable/is_nothrow_move_assignable and so on.